rfcs/0000-kuadrant_crd.md

```diff
@@ -0,0 +1,164 @@
+
+Changing the definition of the `kuadrant` custom resource is relatively straight forward. It is
+intended that the way to enable or disable the support for either policy is not a boolean value. Using
+an enumeration, i.e. `disabled` or `enabled`, enables growing the set of values later.
+
+## Installation
+
+On initial installation, honoring one or the other setting would be fairly straight forward as well:
+
+The both CRDs, as well as their reconcilers all get registered whether the `policy` support is
+`enabled` or `disabled`.
+
+- `enabled` The Kuadrant controller makes sure the external authorizer gets registered with Istio and
+  makes sure there is an `Authorino` and/or `Limitador` object without any changes as it does today.
+- `disabled` None of the above would happen. Most importantly, any missing part required for the above
+  shouldn't have the reconciliation fail and prevent further work it requires to do (as reconciling
+  another policy that's being `enabled`).
+
+Additionally a `ValidatingWebhookConfiguration` gets registered for each CR's `CREATE` operation for
+the entire cluster. The webhook will only allow admissions of resources for which the policy is
+currently enabled. `UPDATE` and `DELETE` should remain unhindered, as in order to transition the
+policy support requires no policies to be present.
+
+If any of the following steps fails, all previously applied changes need to be unrolled and the
+`status` block of the CR would clearly reflected what changed and that Kuadrant is currently _not_
+operational. This is to be considered an atomic operation, either all gets successfully deployed or
+nothing is.
+
+If a policy is disabled but a policy is being added, when disallowing the admission, the webhook would
+use a response code `501 Not Implemented` with the additional `message` indicating the policy is
+disabled and as such can't be honored.
+
+### In an OSSM cluster
+
+The deployment of `Kuadrant` should fail if `authentication` isn't explicitly disabled.
+
+## State transitions of policy support
+
+If the `Kuadrant` object gets updated to change the support of a policy, the change would only be
+considered if no such policy exists in the cluster. No state transitions with regards to policy
+support is possible if policies of the given type are present.
+
+While it would be good to be able to enable the support for a policy while such policies already
+exist, there is much to consider as to what this would entail - here are a few things to consider:
+
+- can we apply all these policies?
+- what ordering should be considered to apply them?
+- what would a conflict mean?
+- what would disabling `authentication` for a cluster, while existing routes are currently protected by these?
+
+Requiring to first remove all policies and then only safely disable or enable the support for that
+kind of policy seems reasonable at this stage and can always be added later, which clearly specified
```

Comment:
Agree so the questions above may still be valid for discussion but I think as a starting point this is simple set of states to deal with initially but also agree that we shouldn't rule this out in the future as likely the flow a user will follow here is ```kubectl delete -f AuthPolicy.yaml ``` enable ```kubectl create -f AuthPolicy.yaml ``` which from an end user perspective will feel somewhat redundant.